cli/tests/test_global_opts.rs

```diff
@@ -16,8 +16,6 @@ use std::ffi::OsString;
 
 use crate::common::{get_stderr_string, TestEnvironment};
 
-pub mod common;
-
 #[test]
```

Comment:
This might be a false alarm, but it seems to me that this PR causes the `jj-cli::runner test_global_opts::test_no_subcommand` test to use the system `jj` instead of `target/debug/jj` half the time. This seems non-deterministic.

I'm running `cargo insta test --test-runner nextest --workspace --accept`, which is essentially `cargo nextest run`. I'm guessing that whether the problem occurs depends on the order of test execution.

The reason I noticed is that my system jj has https://github.com/martinvonz/jj/pull/2034 baked-in, which causes this test to fail unless it's appropriately modified. I guess a better way to debug would be to create `/tmp/bug/jj` that always panics and run `PATH=/tmp/bug:$PATH cargo nextest run`, which I'll try to do.

Reply:
Never mind, that has nothing to do with your PR. The problem occurs when running tests in a merge commit.

Reply:
i haven't been able to replicate that failure. i do get occasional flakes from `test_gc`, but i would be surprised if that's related to this change?
```
--- STDERR:              jj-lib::runner test_git_backend::test_gc ---
thread 'test_git_backend::test_gc' panicked at lib/tests/test_git_backend.rs:151:5:
assertion `left == right` failed
  left: {CommitId("577f8238c5611e318786cad886bfe6b479538687"), CommitId("7d52d6a9cebe821d379c2d1d5a707f04a3dd25f2")}
 right: {CommitId("da30a5e5f1c460afabd6f2594a2a7da8df039d73"), CommitId("577f8238c5611e318786cad886bfe6b479538687"), CommitId("7d52d6a9cebe821d379c2d1d5a707f04a3dd25f2")}
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
looking at https://docs.rs/assert_cmd/latest/src/assert_cmd/cargo.rs.html#183-208, it's deterministic; even if the binary doesn't exist yet, it returns an error rather than falling back to PATH. so i'm not sure how this could be related.

Reply:
The problem also has nothing to do with the system `jj`. Sometimes (non-deterministically), jj's version includes a concatenation of the merge commit's parents' commit ids. I'm guessing that happens when `build.rs` uses git instead of jj to get the version. I have no idea why it's non-deterministic.

Reply:
I created a separate bug for this: https://github.com/martinvonz/jj/issues/2922